tests/ci/integration/run_socat_integration.sh

```diff
@@ -0,0 +1,48 @@
+#!/bin/bash -exu
+# Copyright Amazon.com, Inc. or its affiliates. All Rights Reserved.
+# SPDX-License-Identifier: Apache-2.0 OR ISC
+
+source tests/ci/common_posix_setup.sh
+
+# Set up environment.
+# SRC_ROOT(aws-lc)
+#  - SCRATCH_FOLDER
+#    - SOCAT_SRC
+#    - AWS_LC_BUILD_FOLDER
+#    - AWS_LC_INSTALL_FOLDER
+
+# Assumes script is executed from the root of aws-lc directory
+SCRATCH_FOLDER=${SRC_ROOT}/"scratch"
+AWS_LC_BUILD_FOLDER="${SCRATCH_FOLDER}/aws-lc-build"
+AWS_LC_INSTALL_FOLDER="${SCRATCH_FOLDER}/aws-lc-install"
+SOCAT_SRC="${SCRATCH_FOLDER}/socat"
+
+function build_and_test_socat() {
+  cd "$SOCAT_SRC"
```

Comment:
Pedantic-NP: My preference is that when a specific directory is needed a function does a `pushd "$SOCAT_SRC"` at the beginning, then `popd` at the end, to avoid affecting the callers working directory.